app/controllers/course/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+class Course::UsersController < Course::ModuleController
+  load_and_authorize_resource :course_user, through: :course, parent: false
+  before_action :ensure_not_registered, only: [:create]
+
+  def index # :nodoc:
+  end
+
+  def create # :nodoc:
+    @course_user.workflow_state = :approved if current_course_user && current_course_user.staff?
+    if @course_user.save
+      success = t('.success', role: t("course.users.role.#{@course_user.role}"))
+      redirect_to course_path(@course), success: success
+    else
+      redirect_to course_users_path(@course), danger: @course_user.errors
+    end
+  end
+
+  def update # :nodoc:
+    @course_user.update(course_user_params)
+    if @course_user.save
+      success = t('.success', role: t("course.users.role.#{@course_user.role}"))
+      redirect_to course_users_path(@course), success: success
+    else
+      redirect_to course_users_path(@course), danger: @course_user.errors
+    end
+  end
+
+  def destroy # :nodoc:
+    if @course_user.destroy
+      success = t('.success', role: @course_user.role, email: @course_user.user.email)
+      redirect_to course_users_path(@course), success: success
+    else
+      render 'edit'
+    end
+  end
+
+  private
+
+  def course_user_params # :nodoc:
+    params.require(:course_user).permit(:user_id, :role, :name)
+  end
+
+  def ensure_not_registered # :nodoc:
+    if current_course_user && @course_user.user == current_course_user.user
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.